### PR TITLE
Simplify CI setup

### PR DESCRIPTION
**Pull request by jeromekelleher**
Simplify CI by pulling out nox, and running tests directly.

Add a simple Black action to keep the code well formatted.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
I think this is a significant improvement @astheeggeggs, makes things quite a bit simpler without nox, and we're running Black properly as part of CI to avoid formatting discussions.

What do you think?

**Comment by jeromekelleher** [comment]
Ping @szhan - this makes changes across quite a few files, so best to wait until it's merged before embarking on any significant refactoring

**Comment by astheeggeggs** [comment]
Great - very happy to be guided by you here @jeromekelleher.

**Review by astheeggeggs — APPROVED** [review]

**Pull request merged**
